doc/guides/debugging/probes.rst

```diff
@@ -137,6 +137,31 @@ As with all OpenSDA debug probes, the steps for programming the firmware are:
    should see two USB devices enumerate: a CDC device (serial port) and a
    vendor-specific device (debug port).
 
+.. _stlink-v21-onboard-debug-probe:
+
+ST-LINK/V2-1 Onboard Debug Probe
+********************************
+
+ST-LINK/V2-1 is a serial and debug adapter built into all Nucleo and Discovery
+boards. It provides a bridge between your computer (or other USB host) and the
+embedded target processor, which can be used for debugging, flash programming,
+and serial communication, all over a simple USB cable.
+
+It is compatible with the following host debug tools:
+
+- :ref:`openocd-debug-host-tools`
+- :ref:`jlink-debug-host-tools`
```

Comment:
I think pyOCD recently added support for ST-Link

Reply:
Indeed, but not widely deployed so far. This is worth adding a comment anyway.